src/components/GoalPage/GoalPage.tsx

```diff
@@ -191,6 +177,34 @@ export const GoalPage = ({ user, ssrTime, params: { id } }: ExternalPageProps<{
         [goal, invalidateFn, addParticipantMutation],
     );
 
+    const onAssigneeChange = useCallback(
+        async (activity?: ActivityByIdReturnType) => {
+            if (goal && activity?.user && goal.project) {
+                setCurrentOwner(activity.user);
+                try {
+                    await update({
+                        ...(goal as unknown as GoalUpdate),
+                        owner: {
+                            id: activity.id,
+                            user: {
+                                ...activity.user,
+                            },
+                        },
+                        // Need to pad fields due to zod validation
+                        parent: {
+                            ...goal.project,
+                        },
+                        estimate: goal._lastEstimate,
```

Comment:
This is optional https://github.com/taskany-inc/issues/blob/main/src/schema/goal.ts#L145-L152